tendrl/monitoring_integration/grafana/webhook_receiver.py

```diff
@@ -20,22 +21,20 @@ def __init__(self):
     def _application(self, env, start_response):
         try:
             if env['PATH_INFO'] != '/grafana_callback':
-                start_response(
-                    '404 Not Found',
-                    [('Content-Type', 'text/html')]
-                )
-                response = [b'<h1>Alert Not Found</h1>']
+                response = Response('Alert not found')
+                response.headers['content-length'] = len(response.data)
+                response.status_code = 404
             else:
-                data = env['wsgi.input'].read()
+                data = env['wsgi.input'].read(
+                    int(env['CONTENT_LENGTH'])
+                )
                 data = json.loads(data)
                 self.alert_handler.handle_alert(
                     data["ruleId"]
```

Comment:
Can we put a check for ruleId (For edge cases), or the ruleId will be there no matter what?

Reply:
ok i will do it